File: FreeCodeCamp/Data Structures/Set Class/setClass.py
class Set:
    def __init__(self):
        self.dictionary = {}
        
    def has(self, element):
        return element in self.dictionary
        
    def values(self):
        return self.dictionary.values()
        
    def add(self, element):
        if not self.has(element):
            self.dictionary[element] = element
            return True
        return False
        
    def delete(self, element):
        if self.has(element):
            del self.dictionary[element]
            return True
        return False
        
    def size(self):
        return len(self.dictionary)

File: FreeCodeCamp/Data Structures/Set Class/setClass.js
class Set {
  constructor() {
    // Dictionary will hold the items of our set
    this.dictionary = {};
    this.length = 0;
  }

  // This method will check for the presence of an element and return true or false
  has(element) {
    return this.dictionary[element] !== undefined;
  }

  // This method will return all the values in the set
  values() {
    return Object.values(this.dictionary);
  }

  add(element) {
    if (!this.has(element)) {
      this.dictionary[element] = element;
      this.length++;
      return true;
    }
    return false;
  }

  remove(element) {
    if (this.has(element)) {
      delete this.dictionary[element];
      this.length--;
      return true;
    }

    return false;
  }

  size() {
    return this.length
  }

  union(setB) {
    let aset = new Set()
    for (let a in this.dictionary) aset.add(a)
    for (let b in setB.dictionary) aset.add(b)
    return aset
  }
}



